src/coreclr/System.Private.CoreLib/src/System/Reflection/MdFieldInfo.cs

```diff
@@ -33,8 +33,7 @@ internal override bool CacheEquals(object? o)
             return
                 o is MdFieldInfo m &&
                 m.m_tkField == m_tkField &&
-                m_declaringType.TypeHandle.GetModuleHandle().Equals(
-                    m.m_declaringType.TypeHandle.GetModuleHandle());
+                m_declaringType.Equals(m.m_declaringType);
```

Comment:
This should be also ReferenceEquals. Equals is virtual method - I am not sure whether the JIT is going to be able to de-virtualize it.